src/adapters/rhythmone.js

```diff
@@ -0,0 +1,360 @@
+      "cur": [currency],
+      "id": utils.generateUUID(),
+      "imp":[]
+    };
+    
+    setIfPresent(o.site, "page", function(){
+      var l;
+      try{l = global.top.document.location.href.toString();}
+      catch(ex){l = document.location.href.toString();}
+      return l;
+    });
+    setIfPresent(o.site, "domain", function(){
+      var d = document.location.ancestorOrigins;
+      if(d && d.length > 0)
+        return d[d.length-1];
+      return global.top.document.location.hostname;
+    });
+    setIfPresent(o.site, "name", function(){return global.top.document.title;});
+    
+    o.device.devicetype = ((/(ios|ipod|ipad|iphone|android)/i).test(global.navigator.userAgent) ? 1 : ((/(smart[-]?tv|hbbtv|appletv|googletv|hdmi|netcast\.tv|viera|nettv|roku|\bdtv\b|sonydtv|inettvbrowser|\btv\b)/i).test(global.navigator.userAgent) ? 3 : 2));
+    
+    setIfPresent(o.device, "h", function(){return global.screen.height;});
+    setIfPresent(o.device, "w", function(){return global.screen.width;});
+    
+    for(var i = 0; i<bids.length; i++){
+      var bidID = utils.generateUUID();
+      slotMap[bidID] = bids[i];
+      slotMap[bids[i].placementCode] = bids[i];
+      
+      if(bidParams.method === "post")
+        track(debug, 'hb', 'bidRequest');
+        
+      for(var j = 0; j<bids[i].sizes.length; j++){
+        o.imp.push({
+          "id": bidID,
+          "tagId": bids[i].placementCode,
+          "bidfloor": bidfloor,
+          "bidfloorcur": currency,
+          "banner": {
+            "id": utils.generateUUID(),
+            "pos": 0,
+            "w": bids[i].sizes[j][0],
+            "h": bids[i].sizes[j][1]
+          }
+        });
+      }
+    }
+    
+    return o;
+  }
+  
+  this.callBids = function(params){
+
+    var slotMap = {},
+      bidParams = getBidParameters(params.bids);
+  
+    debug = (bidParams !== null && bidParams.debug === true);
+  
+    track(debug, 'hb', 'callBids');
+
+    if(bidParams === null){
+      noBids(params);
+      track(debug, 'hb', 'misconfiguration');
+      return;
+    }
+
+    // default to GET request
+    if(typeof bidParams.method !== "string")
+      bidParams.method = "get";
+    
+    bidParams.method = bidParams.method.toLowerCase();
+    
+    sniffAuctionEnd();
+
+    track(debug, 'hb', 'rmpRequest');
+
+    var ortbJSON = getORTBJSON(params.bids, slotMap, bidParams);
+    
+    load(bidParams, getRMPURL(bidParams, ortbJSON, params.bids), JSON.stringify(ortbJSON), function(code, msg, txt){
+    
+      if(auctionEnded === true)
+        return;
+    
+      requestCompleted = true;
+    
+      logToConsole(txt);
+    
+      var bidCount = 0;
+    
+      if(code === -1)
+        track(debug, 'hb', 'rmpReplyFail', msg);
+      else{
+        try{
+          var result = JSON.parse(txt),
+            registerBid = function(bid){
+            
+              slotMap[bid.impid].success = 1;
+              
+              var pbResponse = bidfactory.createBid(1),
```

Comment:
Adapters can now pass a constant for status and the originating bid request, the first for readability and the second to match bid request / response IDs per #509.

e.g. 
```
var bid = bidfactory.createBid(CONSTANTS.STATUS.GOOD, bidRequest);
```